typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -789,7 +845,7 @@
             (match (resolve ty)
               [(Function: _) #t]
               [(Union: _ elems) (andmap loop elems)]
-              [(Intersection: elems) (ormap loop elems)]
+              [(Intersection: elems _) (ormap loop elems)]
```

Comment:
Okay, so I'm fixing the other ones -- but here we're ensuring a type is a function... and if it's a function, it should obviously be so, not only in a refinement prop. So I think it's fine to ignore the proposition here (it's sound and the argument for "complete" here seems silly)